### PR TITLE
fix input-item textarea-item search-bar bug #1955

### PR DESCRIPTION
**Pull request by pingan1927**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1956)

### CODE REVIEW

**Comment by pingan1927** [comment]
浏览器存在input触发blur事件又自行触发了focus事件，（但虚拟键盘已经收起），这个事情再次点击input弹出键盘输入内容，并不会触发focus事件。

看代码稍微理解就清楚问题所在。

File: components/input-item/index.web.tsx [hunk 1]
```diff
@@ -124,9 +124,11 @@ class InputItem extends React.Component<InputItemProps, any> {
...
     this.debounceTimeout = setTimeout(() => {
-      this.setState({
-        focus: false,
-      });
+      if (document.activeElement !== (this.refs.input as any).refs.input) {
+        this.setState({
+          focus: false,
+        });
+      }
     }, 200);
```
Comment (pingan1927) [inline components/input-item/index.web.tsx]: 问题就在于延迟200ms执行，如果浏览器自动触发了focus，且一直处于focus状态，但是state又被这行代码设置为false，导致跟预期不符。

**Comment by warmhug** [comment]
测好了吗？没问题尽快合并，并 修改到 master 分支上

**Comment by silentcloud** [comment]
@pingan1927 你这边测试ok的？我直接合了

**Comment by pingan1927** [comment]
我今天重新部署试试看业务是否修复OK，验证完毕合并到2.0

**Comment by pingan1927** [comment]
业务验证OK，今天推一个pr到2.0

**Comment by silentcloud** [comment]
@pingan1927 我已经手动合到 2.0 了，目前2.0已经有了

**Comment by pingan1927** [comment]
good

**Pull request merged**